c2rust-transpile/src/convert_type.rs

```diff
@@ -301,30 +303,28 @@ impl TypeConverter {
     ) -> TranslationResult<Box<Type>> {
         if self.translate_valist && ctxt.is_va_list(ctype) {
-            let path = vec![std_or_core(self.emit_no_std), "ffi", "VaList"];
-            let ty = mk().path_ty(mk().abs_path(path));
-            return Ok(ty);
+            return Ok(path![::core::ffi::VaList]);
         }
 
         match ctxt.index(ctype).kind {
-            CTypeKind::Void => Ok(mk().tuple_ty(vec![] as Vec<Box<Type>>)),
-            CTypeKind::Bool => Ok(mk().path_ty(mk().path(vec!["bool"]))),
-            CTypeKind::Short => Ok(mk().path_ty(mk().path(vec!["libc", "c_short"]))),
-            CTypeKind::Int => Ok(mk().path_ty(mk().path(vec!["libc", "c_int"]))),
-            CTypeKind::Long => Ok(mk().path_ty(mk().path(vec!["libc", "c_long"]))),
-            CTypeKind::LongLong => Ok(mk().path_ty(mk().path(vec!["libc", "c_longlong"]))),
-            CTypeKind::UShort => Ok(mk().path_ty(mk().path(vec!["libc", "c_ushort"]))),
-            CTypeKind::UInt => Ok(mk().path_ty(mk().path(vec!["libc", "c_uint"]))),
-            CTypeKind::ULong => Ok(mk().path_ty(mk().path(vec!["libc", "c_ulong"]))),
-            CTypeKind::ULongLong => Ok(mk().path_ty(mk().path(vec!["libc", "c_ulonglong"]))),
-            CTypeKind::SChar => Ok(mk().path_ty(mk().path(vec!["libc", "c_schar"]))),
-            CTypeKind::UChar => Ok(mk().path_ty(mk().path(vec!["libc", "c_uchar"]))),
-            CTypeKind::Char => Ok(mk().path_ty(mk().path(vec!["libc", "c_char"]))),
-            CTypeKind::Double => Ok(mk().path_ty(mk().path(vec!["libc", "c_double"]))),
-            CTypeKind::LongDouble => Ok(mk().path_ty(mk().path(vec!["f128", "f128"]))),
-            CTypeKind::Float => Ok(mk().path_ty(mk().path(vec!["libc", "c_float"]))),
-            CTypeKind::Int128 => Ok(mk().path_ty(mk().path(vec!["i128"]))),
-            CTypeKind::UInt128 => Ok(mk().path_ty(mk().path(vec!["bf16"]))),
+            CTypeKind::Void => Ok(mk().tuple_ty(vec![])),
+            CTypeKind::Bool => Ok(path![bool]),
+            CTypeKind::Short => Ok(path![::libc::c_short]),
+            CTypeKind::Int => Ok(path![::libc::c_int]),
+            CTypeKind::Long => Ok(path![::libc::c_long]),
+            CTypeKind::LongLong => Ok(path![::libc::c_longlong]),
+            CTypeKind::UShort => Ok(path![::libc::c_ushort]),
+            CTypeKind::UInt => Ok(path![::libc::c_uint]),
+            CTypeKind::ULong => Ok(path![::libc::c_ulong]),
+            CTypeKind::ULongLong => Ok(path![::libc::c_ulonglong]),
+            CTypeKind::SChar => Ok(path![::libc::c_schar]),
+            CTypeKind::UChar => Ok(path![::libc::c_uchar]),
+            CTypeKind::Char => Ok(path![::libc::c_char]),
+            CTypeKind::Double => Ok(path![::libc::c_double]),
+            CTypeKind::LongDouble => Ok(path![::f128::f128]),
+            CTypeKind::Float => Ok(path![::libc::c_float]),
+            CTypeKind::Int128 => Ok(path![i128]),
+            CTypeKind::UInt128 => Ok(path![bf16]),
```

Comment:
Unrelated to your refactor, but this seems like a bug.  Rust has `u128`, which I think it should be, and `bf16` should correspond to `bfloat16`.

Reply:
Indeed it is.

Reply:
Filed an issue for it to fix in a bit: https://github.com/immunant/c2rust/issues/486.